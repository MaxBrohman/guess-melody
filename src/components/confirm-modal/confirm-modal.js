import React from 'react';

// eslint-disable-next-line
export const ConfirmModal = ({ onConfirm, onClose }) => (
  <section className="modal">
    <button className="modal__close" type="button" onClick={onClose}>
      <span className="visually-hidden">Закрыть</span>
    </button>
    <h2 className="modal__title">Подтверждение</h2>
    <p className="modal__text">Вы уверены что хотите начать игру заново?</p>
    <div className="modal__buttons">
      <button className="modal__button button" onClick={onConfirm}>Ок</button>
      <button className="modal__button button" onClick={onClose}>Отмена</button>
    </div>
  </section>
);
