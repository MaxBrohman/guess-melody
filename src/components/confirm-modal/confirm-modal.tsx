import React from 'react';

type TConfirmModalProps = {
  onConfirm: () => void;
  onClose: () => void;
}

export const ConfirmModal = ({onConfirm, onClose}: TConfirmModalProps): JSX.Element => (
  <section className="modal">
    <button className="modal__close" type="button" onClick={onClose}>
      <span className="visually-hidden">Закрыть</span>
    </button>
    <h2 className="modal__title">Подтверждение</h2>
    <p className="modal__text">Вы уверены что хотите начать игру заново?</p>
    <div className="modal__buttons">
      <button className="modal__button button" onClick={onConfirm}>Ок</button>
      <button className="modal__button button" onClick={onClose}>Отмена</button>
    </div>
  </section>
);
