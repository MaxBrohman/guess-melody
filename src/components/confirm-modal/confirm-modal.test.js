import React from 'react';
import renderer from 'react-test-renderer';

import {ConfirmModal} from './confirm-modal';

it(`Renders ConfirmModal component correctly`, () => {
  const onConfirm = jest.fn();
  const onClose = jest.fn();
  const tree = renderer
                .create(
                    <ConfirmModal
                      onConfirm={onConfirm}
                      onClose={onClose}
                    />
                )
                .toJSON();
  expect(tree).toMatchSnapshot();
});
