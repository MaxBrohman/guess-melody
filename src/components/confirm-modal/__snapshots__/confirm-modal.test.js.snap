// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ConfirmModal component correctly 1`] = `
<section
  className="modal"
>
  <button
    className="modal__close"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="visually-hidden"
    >
      Закрыть
    </span>
  </button>
  <h2
    className="modal__title"
  >
    Подтверждение
  </h2>
  <p
    className="modal__text"
  >
    Вы уверены что хотите начать игру заново?
  </p>
  <div
    className="modal__buttons"
  >
    <button
      className="modal__button button"
      onClick={[MockFunction]}
    >
      Ок
    </button>
    <button
      className="modal__button button"
      onClick={[MockFunction]}
    >
      Отмена
    </button>
  </div>
</section>
`;
